app/src/main/java/org/mozilla/fenix/crashes/CrashReporterController.kt

```diff
@@ -4,81 +4,78 @@
      */
-    fun handleCloseAndRestore(sendCrash: Boolean): Job? {
-        val job = submitReportIfNecessary(sendCrash)
+    fun handleCloseAndRestore(sendCrashes: Boolean): Job? {
+        val job = submitPendingNonFatalCrashesIfNecessary(sendCrashes)
 
         components.useCases.sessionUseCases.crashRecovery.invoke()
-        navController.popBackStack()
+
         return job
     }
 
     /**
-     * Closes the crash reporter fragment and the tab.
+     * Closes the current tab, restore all sessions and optionally report pending non-fatal crashes.
      *
-     * @param sendCrash If true, submit a crash report.
-     * @return Job if report is submitted through an IO thread, null otherwise
+     * @param reportCrashes If true, submit crash reports for all current non-fatal crashes.
+     * @return [Job] allowing to check status / cancel the reporting operation or null if reporting is disabled.
      */
-    fun handleCloseAndRemove(sendCrash: Boolean): Job? {
-        sessionId ?: return null
-        val job = submitReportIfNecessary(sendCrash)
+    fun handleCloseAndRemove(reportCrashes: Boolean): Job? {
+        val job = submitPendingNonFatalCrashesIfNecessary(reportCrashes)
 
         components.useCases.tabsUseCases.removeTab(sessionId)
         components.useCases.sessionUseCases.crashRecovery.invoke()
 
-        navController.nav(
-            R.id.crashReporterFragment,
-            CrashReporterFragmentDirections.actionGlobalHome()
-        )
-
         return job
     }
 
     /**
-     * Submits the crash report if the "Send crash" checkbox was checked and the setting is enabled.
+     * Submits all pending non-fatal crash reports if the "Send crash" checkbox was checked
+     * and the report crashes setting is enabled.
+     * Also clears the current list of non-fatal crashes irrespective of whether they are reported or not.
      *
-     * @param sendCrash If true, submit a crash report.
-     * @return Job if report is submitted through an IO thread, null otherwise
+     * @param reportCrashes A second condition beside crash reporting being enabled in app settings
+     * based on which the current crashes will be reported or immediately disposed off.
+     * @return [Job] allowing to check status / cancel the reporting operation or null if reporting is disabled.
      */
+    @VisibleForTesting
     @OptIn(DelicateCoroutinesApi::class) // GlobalScope usage
-    private fun submitReportIfNecessary(sendCrash: Boolean): Job? {
+    internal fun submitPendingNonFatalCrashesIfNecessary(reportCrashes: Boolean): Job? {
         var job: Job? = null
-        val didSubmitReport = if (sendCrash && settings.isCrashReportingEnabled) {
+        if (reportCrashes && settings.isCrashReportingEnabled) {
             job = GlobalScope.launch(Dispatchers.IO) {
-                components.analytics.crashReporter.submitReport(crash)
+                val crashes = appStore.state.nonFatalCrashes
+                crashes.forEach {
```

Comment:
(Related to the comment below)  I see, we're not just submitting the crash related to the `sessionId` but reporting all crashes.  Is this something that we want?